### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Comment by miczyg1** [comment]
Some documents are still missing. I wanted to describe the high-level methodology (empty file is created as a placeholder) and some principles behind the classes structure (there is a pattern I have maintained across 4 classes) and how they work. I would like to write it down as well.

File: README.md [hunk 1]
```diff
@@ -9,18 +84,46 @@ Dasharo Openness Score measuring utility
...
-- matplotlib: `pip3 install matplotlib`
+- `make`
 
-## Checking Python style
+Python requirements:
+
+```bash
+virtualenv -p $(which python3) venv
+source venv/bin/activate
+pip install -r requirements.txt
```
Comment (macpijan) [inline README.md:94]: It fails for me with: `glib.h: No such file or directory`

Comment (macpijan) [reply README.md:94]: Fixed by installing `glib2-devel` on Fedora 38

File: docs/source/design.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
+Dasharo Openness Score design
```
Comment (macpijan) [inline docs/source/design.rst:1]: Why did we decide to use another framework for documentation, not mkdocs we typically use?

Comment (macpijan) [reply docs/source/design.rst:1]: You could use this template to have the same look-and-feel as with the proposed solution, if that was desired: https://mkdocs.readthedocs.io/en/restructure-compat/user-guide/styling-your-docs/

Sticking with mkdocs, we only need to use one documentation framework, we can still use markdown, and we can reuse the know-how, pre-commit hooks (which are missing in this repo, BTW), etc.

File: README.md [hunk 3]
```diff
@@ -9,18 +84,46 @@ Dasharo Openness Score measuring utility
...
+```bash
+(venv) make html
+python -m http.server 8000
+```
+
+For live preview run:
 
 ```bash
-pip3 install pycodestyle
+(venv) make livehtml
```
Comment (macpijan) [inline README.md:115]: This documentation should be hosted somewhere - at least GitHub pages. Not necessarily now, but let's create issue, at least.

File: README.md (hunk 1 above)
Comment (macpijan) [reply README.md:94]: Please create issue for better dev environment. cbfstool, and others, should also be in a container, or nix-shell, or something for easy setup and CI runs.

File: docs/source/design.rst (hunk 2 above)
Comment (macpijan) [reply docs/source/design.rst:1]: https://mkdocstrings.github.io/ plugin can put Python docstrings into mkdocs as well

**Review by macpijan — COMMENTED** [review]
Left some first comments. Still reviewing .rst files.

**Review by macpijan — COMMENTED** [review]
Good job overall. The stats and graphs looks nice.

One thing maybe I'm missing: how do we plan to show these results to the client? Let's say I release Dasharo for some platform today and want to present such report. Where do I host/present such report?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,81 @@
...
+Dasharo Openness Score measuring utility.
+
+## What is Dasharo Openness Score
+
+Have you ever wondered how open is your open-source firmware? How much
+closed-source and binary blobs is still there? Dasharo Openness Score utility
+answers those questions.
+
+Dasharo Openness is a report showing the open-source code to closed-source code
+ratio in the Dasharo firmware images. The results are also presented as a pie
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply README.md:12]:
```suggestion
ratio in the firmware images. The results are also presented as a pie
```

File: README.md (hunk 1 above)
Comment (miczyg1) [reply README.md:94]: Too late for that I guess: https://github.com/Dasharo/Openness-Score/pull/6

File: docs/source/design.rst (hunk 2 above)
Comment (miczyg1) [reply docs/source/design.rst:1]: The searches did not give me mkdocs results with docstrings unfortunately thus I went with Sphinx... I guess insufficient research from my side.

**Comment by miczyg1** [comment]
> Good job overall. The stats and graphs looks nice.
> 
> One thing maybe I'm missing: how do we plan to show these results to the client? Let's say I release Dasharo for some platform today and want to present such report. Where do I host/present such report?

I was thinking to add a separate section per board in doc.sdasharo.com in the supported hardware

File: docs/source/design.rst (hunk 2 above)
Comment (macpijan) [reply docs/source/design.rst:1]: Let's create issue with content / link to this discussion. We may consider switch in future revisions.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+__pycache__
+*.bin
+*.rom
+build
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply .gitignore:4]: Subject to #4

[the remark this answers is not in the document]
Comment (miczyg1) [reply .gitignore:4]: https://github.com/Dasharo/Openness-Score/pull/4/commits/9f90797ed4732637d23c3bc86d3bfa7c43f73c4c

File: docs/source/design.rst (hunk 2 above)
Comment (miczyg1) [reply docs/source/design.rst:1]: https://github.com/Dasharo/dasharo-issues/issues/451

File: docs/source/index.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+Dasharo Openness Score utility is capable of parsing Dasharo coreboot-based
+images as well as vendor UEFI images. Thanks to that one can easily compare
+how many bytes of the firmware have been liberated as well as by how much the
+Trusted Computing Base (TCB) has been reduced.
+
+Dasharo Openness Score utility not only support Dasharo coreboot-based images,
+but also many more coreboot distributions like heads.
+
+How does it work?
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply docs/source/index.rst:37]: I don't plan to extend README anymore. As the main page in PShinx looked weirdly empty, I just copied over the README contents there. We should extend the generated documentation from onwards and minimize README modifications

File: README.md (hunk 3 above)
Comment (miczyg1) [reply README.md:115]: Will be solved by https://github.com/Dasharo/dasharo-issues/issues/451 right?

**Pull request merged**